src/image/transform/mask/li.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * see http://rsb.info.nih.gov/ij/developer/source/ij/process/AutoThresholder.java.html
+ * The method is present in: Implements Li's Minimum Cross Entropy thresholding method
+ * This implementation is based on the iterative version (Ref. 2nd reference below) of the algorithm.
+ *  1) Li, CH & Lee, CK (1993), "Minimum Cross 	Entropy Thresholding", Pattern Recognition 26(4): 61 625
+ *  2) Li, CH & Tam, PKS (1998), "An Iterative 	Algorithm for Minimum Cross Entropy Thresholding",
+ *     Pattern 	Recognition Letters 18(8): 771-776
+ *  3) Sezgin, M & Sankur, B (2004), "Survey 	over Image Thresholding Techniques and Quantitative Performance
+ *     Evaluation",Journal of Electronic Imaging 13(1): 146-165
+ * @param histogram - the histogram of the image
+ *        total - the number of pixels in the image
+ * @returns {number} - the threshold
+ */
+
+
+export default function li(histogram, total) {
+
+    let threshold;
+    let sum_back; /* sum of the background pixels at a given threshold */
+    let sum_obj;  /* sum of the object pixels at a given threshold */
+    let num_back; /* number of background pixels at a given threshold */
+    let num_obj;  /* number of object pixels at a given threshold */
+    let old_thresh;
+    let new_thresh;
+    let mean_back; /* mean of the background pixels at a given threshold */
+    let mean_obj;  /* mean of the object pixels at a given threshold */
+    let mean;  /* mean gray-level in the image */
+    let tolerance; /* threshold tolerance */
+    let temp;
+    tolerance = 0.5;
+
+    /* Calculate the mean gray-level */
+    mean = 0.0;
+    for (let ih = 0; ih < histogram.length; ih++) {
+        mean += ih * histogram[ih];
+    }
+
+    mean /= total;
+    /* Initial estimate */
+    new_thresh = mean;
+
+    do {
+        old_thresh = new_thresh;
+        threshold = old_thresh + 0.5;	/* range */
```

Comment:
the java code is casting the value to `(int)`. It very important to do the same in JavaScript if we want the same result. This is possible with the bitwise `or` operator: `threshold = (old_thresh + 0.5) | 0`